File: MvpFramework/src/main/java/com/ytx/mvpframework/presenter/ActivityPresenter.java
package com.ytx.mvpframework.presenter;

import android.support.v4.app.FragmentActivity;

import com.ytx.mvpframework.view.IView;

public class ActivityPresenter<V extends IView> extends LifecyclePresenter<V>  {

    public ActivityPresenter(V view) {
        super(view);
        if (view instanceof FragmentActivity) {
            bindActivity((FragmentActivity) view);
        }
    }

    public final void bindActivity(FragmentActivity activity) {
        bindLifecycle(activity);
    }

    public final void unbindActivity() {
        unbindLifecycle();
    }
}

File: MvpFramework/src/main/java/com/ytx/mvpframework/presenter/ActivityPresenter.kt
//@file:JvmName("ActivityPresenter")
//package com.ytx.mvpframework.presenter
//
//import android.support.v4.app.FragmentActivity
//import com.ytx.mvpframework.view.IView
//
//open class ActivityPresenter<V : IView>(view: V) : LifecyclePresenter<V>(view) {
//
//    init {
//        if (view is FragmentActivity) {
//            bindActivity(view)
//        }
//    }
//
//    public fun bindActivity(activity: FragmentActivity) {
//        bindLifecycle(activity)
//    }
//
//    public fun unbindActivity() {
//        unbindLifecycle()
//    }
//}